cmd/res/configuration.toml

```diff
@@ -0,0 +1,104 @@
+    [Writable.InsecureSecrets.Camera001]
+    path = "credentials001"
+      [Writable.InsecureSecrets.Camera001.Secrets]
+      username = "administrator"
+      password = "Password1!"
+    # If having more than one camera, uncomment the following config settings
+    [Writable.InsecureSecrets.Camera002]
+    path = "credentials002"
+      [Writable.InsecureSecrets.Camera002.Secrets]
+      username = "administrator"
+      password = "Password1!"
+
+[Service]
+HealthCheckInterval = "10s"
+Host = "0.0.0.0"
+Port = 59985
+ServerBindAddr = ""  # blank value defaults to Service.Host value
+StartupMsg = "Camera device service started"
+# MaxRequestSize limit the request body size in byte of put command
+MaxRequestSize = 0 # value 0 unlimit the request size.
+RequestTimeout = "5s"
+
+# Only used when EDGEX_SECURITY_SECRET_STORE=true which is now required for secure consul
+[SecretStore]
+Type = "vault"
+Host = "localhost"
+Port = 8200
+Path = "device-onvif-camera/"
+Protocol = "http"
+RootCaCertPath = ""
+ServerName = ""
+TokenFile = "/tmp/edgex/secrets/device-onvif-camera/secrets-token.json"
+  [SecretStore.Authentication]
+  AuthType = "X-Vault-Token"
+
+[Registry]
+Type = "consul"
+Host = "localhost"
+Port = 8500
+
+[Clients]
+  [Clients.core-data]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59880
+
+  [Clients.core-metadata]
+  Protocol = "http"
+  Host = "localhost"
+  Port = 59881
+
+[MessageQueue]
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+PublishTopicPrefix = "edgex/events/device" # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
+  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-onvif-camera"
+  # Connection information
+  Qos = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+
+[Device]
+DataTransform = true
+MaxCmdOps = 128
+MaxCmdValueLen = 256
+ProfilesDir = "./res/profiles"
+DevicesDir = "./res/devices"
+UpdateLastConnected = false
+EnableAsyncReadings = true
+AsyncBufferSize = 16
+Labels = []
+UseMessageBus = true
+  [Device.Discovery]
+  Enabled = false
+  Interval = "30s"
+
+# Driver configs
+[Driver]
+CredentialsRetryTime = "120" # Seconds
+CredentialsRetryWait = "1" # Seconds
+RequestTimeout = "5" # Seconds
+DiscoveryEthernetInterface = "en0"
+DefaultAuthMode = "usernametoken"
+DefaultSecretPath = "credentials001"
+# BaseNotificationURL indicates the device service network location
+BaseNotificationURL = "http://192.168.12.112:59985"
```

Comment:
This will always be different and require this to always be update when deployed.
The service can and should dynamically figure out its IP.
This can be accomplished using ` net.Interfaces()` and finding the address with name `eth0` 
Example of running `ifconfig` in running container for Device Vitual
```
/ $ ifconfig
eth0      Link encap:Ethernet  HWaddr 02:42:AC:1A:00:0C  
          inet addr:172.26.0.12  Bcast:172.26.255.255  Mask:255.255.0.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:109991 errors:0 dropped:0 overruns:0 frame:0
          TX packets:164209 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:0 
          RX bytes:9283272 (8.8 MiB)  TX bytes:63810690 (60.8 MiB)

lo        Link encap:Local Loopback  
          inet addr:127.0.0.1  Mask:255.0.0.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:112 errors:0 dropped:0 overruns:0 frame:0
          TX packets:112 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:7440 (7.2 KiB)  TX bytes:7440 (7.2 KiB)
```

Reply:
This improvement needs more time to investigate and implement, open the issue https://github.com/edgexfoundry-holding/device-onvif-camera/issues/38.

Reply:
@lenny-intel this will only get the docker-network ip address. @weichou1229 isn't this value supposed to your machine's ip address as seen by your local network such that ip-cameras can access it via that IP?

Reply:
@ajcasagrande, yes you are correct. The container IP address is accessible for the host system, but not an external dev ice on the network, like a camera. :-(

Reply:
One approach could be setting and environment variable like HOST_IP prior to running cocker-compose and use it as the override value off this setting:
```
export HOST_IP=$(ifconfig eth0 | grep "inet " | awk '{ print $2 }')
docker-compose -p edgex up -d
```
in the compose file have:
```
environment:
   DRIVER_BASENOTIFICATIONURL: http://${HOST_IP}
```

I have verified this override works using `SERVICE_STARTUPMSG: http://${HOST_IP}` on Device Virtual

Reply:
PR opened https://github.com/edgexfoundry-holding/device-onvif-camera/pull/41

Reply:
PR opened for the doc https://github.com/edgexfoundry-holding/device-onvif-camera/pull/46